blackbox/docs/sql/administration/ingestion_rules.txt

```diff
@@ -0,0 +1,92 @@
+.. highlight:: psql
+.. _administration-ingestion-rules:
+
+===============
+Ingestion Rules
+===============
+
+In CrateDB, ingestion rules define where and how the incoming data of a
+specific ingestion should be routed and stored.
+
+In this section, we will learn about creating and deleting ingestion rules,
+listing the existing ingestion rules and finally the required privileges for
+the ``CREATE`` and ``DROP INGEST RULE`` statements, when the
+:ref:`administration_user_management` feature is enabled.
+
+Creating and Deleting Ingestion Rules
+.....................................
+
+Suppose we have installed an ingestion implementation for an ``mqtt`` broker,
+and we would like to filter temperature data streams by topic and store them to
+``temp_data.raw``::
+
+    cr> CREATE INGEST RULE temperature
+    ... ON mqtt
+    ... WHERE topic like 'temperature/%'
+    ... INTO temp_data.raw;
+    CREATE OK, 1 row affected (... sec)
+
+It is possible to create and drop ingestion rules even if the corresponding
```

Comment:
Maybe this could be phrased in a way that suggests that this *could* change? I think offering some basic validation that the rule is sensible would be an improvement that we might want to implement at some point

Reply:
I can imagine that this  otherwise forces users to use a "trial and error" approach to get the rule right.

Reply:
Well, this is a chicken and egg problem isn't it? We don't want to miss incoming data so we'd want the rule to be there before the implementation is enabled. We don't enforce any sort of unicity on having just one rule/implementation so the user can experiment with different rules for the same implementation. 
I think we will not change this (but can possibly augment it at some point with a rule testing tool or such?!) but this is more of a product decision.
Configuration is generally speaking trial and error so I don't see these rules as something different. @dobe @joemoe Please let us know if you think differently.

Reply:
The ingest-source implementation needs to be avaiable when a Crate node is launched. So once a rule is created it would be possible to do a basic validation. In the example here it could figure out if `topic` is a valid identifier.

Yes configuration is a bit of trial and error - I think the point is about how immediate the feedback is. For example if you use  `SET` you get an immediate error if the setting name is invalid. Now with rules you can create a rule that is accepted but might end up filling your log with hundreds of errors - depending on how the whereClause is evaluated and what happens if it's invalid. 

I'm not suggesting to change the behaviour now - I just would like us to fully commit to it now.